### PR TITLE
Consistently format enums, fix non-identifier IDs.

### LINKED ISSUE #1519: Consistently format semantic convention enums
Semantic convention enums currently have inconsistent formatting. For example, `os.type` values are all capital (https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/os.yaml#L10) whereas `cloud.infrastructure_service` values are underscored lowercase (https://github.com/open-telemetry/opentelemetry-specification/blob/main/semantic_conventions/resource/cloud.yaml#L45).

With the new requirement of [autogenerating the semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/pull/1515), formatting in the YAML files is important for the code generator developers. Bless a format and consistently use it everywhere.

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #1519 (again) and also fixes some `id`s containing spaces (`Alibaba Cloud` -> `alibaba_cloud`) or starting with numbers (`1xRTT` -> `csma2000_1xrtt`) which tripped up code generators (open-telemetry/build-tools#59).

## Changes

Changes enum IDs to be lowercase.

Also changes values and briefs of `net.host.connection.subtype` and `rpc.grpc.status_code` to match the IDs. http.flavor is left unchanged since it is already quite old (but http.flavor should probably be refactored anyway separately).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ release.
 
 ### Semantic Conventions
 
+- BREAKING: Change enum member IDs to lowercase without spaces, not starting with numbers.
```
Comment (Oberon00) [inline CHANGELOG.md:25]: This is a breaking change, but to an experimental part. The oldest changed enum value is from #1156, Oct 30, 2020.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
@Oberon00 Regarding the breaking portion, is it enough to let Maintainers know next Monday, or should we do something else?

**Comment by Oberon00** [comment]
I think that would be enough, as actual data breakage is only in stuff that is new *and* recent. But probably I'm the wrong person to ask that question to. 😃

**Comment by carlosalberto** [comment]
@tigrannajaryan maybe you have something to say?

**Comment by carlosalberto** [comment]
@Oberon00 Please update the CHANGELOG so we can merge ;)

**Pull request merged**